# [LAB-07] 6-일원분산분석 - 연습문제

## 준비작업

### 라이브러리 참조

In [1]:
from jussam import load_data
from helpers import *

📦 연세대학교 주영아 교수가 제작한 라이브러리를 사용중입니다.
📧 Email(1): j.purplerose@yonsei.ac.kr
📧 Email(2): j.purplerose@gmail.com
📝 Website: https://juyounga.kr/


## 문제 1

reading 데이터셋은 유아들을 대상으로 세 가지 읽는 방법을 비교,실험하여 얻은 독서평가 점수 데이터이다.

세 방법에 통계적으로 유의한 차이가 있는지 검정하라.

In [2]:
df1 = load_data("reading")
df1.head()

📚 유아들을 대상으로 세 가지 읽는 방법을 비교,실험하여 얻은 독서평가 점수 데이터 (출처: 방송통신대학교 통계학개론)


,독서방법,점수
0,B,7.700
1,B,1.500
2,B,4.300
3,B,5.400
4,B,4.300


In [3]:
my_stats.anova_oneway(df1, y='점수', between='독서방법')

,test,Source,ddof1,ddof2,F,p_unc,np2,effect_size
0,welch_anova,독서방법,2,48.204,10.787,0.000,0.283,Large


### 💡인사이트

- **가정 확인**: `anova_oneway` 함수는 분석에 앞서 정규성(normaltest)과 등분산성 검정을 수행한다. 결과표의 `test`가 `welch_anova`로 출력되었다는 것은 **등분산성 가정이 위배(불충족)**되어, 등분산을 가정하지 않는 **Welch-ANOVA**가 자동 선택되었음을 의미한다. (등분산이 충족되면 일반 `anova`가 선택된다.)
- **검정 결과**: F(2, 48.204) = 10.787, p ≈ 0.000 으로 유의수준 0.05보다 작다. 따라서 "세 독서방법의 평균 점수가 모두 같다"는 귀무가설을 **기각**한다. 즉, **세 가지 읽는 방법에 따른 독서평가 점수에는 통계적으로 유의한 차이가 있다.**
- **효과크기**: 부분 에타제곱(np2) = 0.283 으로 0.14 이상에 해당하여 **큰 효과(Large)**로 해석된다. 독서방법이 점수 분산의 약 28.3%를 설명한다.
- **결론**: 독서방법은 독서평가 점수에 유의하고 큰 영향을 미친다. 다만 분산분석은 "어느 집단들 사이에" 차이가 있는지는 알려주지 않으므로, 구체적인 집단 쌍 간 차이를 확인하려면 사후검정(`posthoc_oneway`)이 필요하다.


## 문제 2

서울시에서 정의하는 상권의 종류는 발달상권, 골목상권, 관광특구, 전통시장이 있다.

commercial 데이터셋은 서울시의 2023년 상권에 따른 분기별 매출액과 매출건수에 대한 자료이다.

상권 종류별 2023년 한 해 동안의 평균 매출액은 어떤 차이가 보이는지 분석하라.

주어진 데이터는 서울시에서 제공하는 집계가 완료된 데이터이므로 모든 결측치는 없는 상태이며 이상치도 실제 매출액 데이터이므로 별도의 정제 과정 없이 진행하는 것으로 한다.

In [4]:
df2 = load_data("commercial")
df2.head()

📚 서울시의 2023년 상권에 따른 분기별 매출액과 매출건수에 대한 자료 (출처: 서울 열린데이터 광장)


,quarter,code,type,name,sales_amount,number_of_sales
0,1,3001491,관광특구,이태원 관광특구,72523644100,2496865
1,1,3001492,관광특구,명동 남대문 북창동 다동 무교동 관광특구,283332346393,10246122
2,1,3001493,관광특구,동대문패션타운 관광특구,81696730221,2880324
3,1,3001494,관광특구,종로·청계 관광특구,212062656625,4960006
4,1,3001495,관광특구,잠실 관광특구,231338386876,5995166


In [5]:
my_stats.anova_oneway(df2, y='sales_amount', between='type')

,test,Source,ddof1,ddof2,F,p_unc,np2,effect_size
0,welch_anova,type,3,110.239,161.433,0.000,0.228,Large


### 💡인사이트

- **가정 확인**: `anova_oneway` 함수가 정규성과 등분산성 검정을 수행한 뒤 `test`를 `welch_anova`로 출력하였다. 이는 상권 종류별 매출액의 **등분산성 가정이 위배**되어 **Welch-ANOVA**가 자동 선택되었음을 의미한다. (문제 조건상 결측치·이상치 정제는 생략하였다.)
- **검정 결과**: F(3, 110.239) = 161.433, p ≈ 0.000 으로 유의수준 0.05보다 작다. 따라서 "네 가지 상권 종류(발달상권·골목상권·관광특구·전통시장)의 평균 매출액이 모두 같다"는 귀무가설을 **기각**한다. 즉, **상권 종류에 따라 2023년 평균 매출액에는 통계적으로 유의한 차이가 존재한다.**
- **효과크기**: 부분 에타제곱(np2) = 0.228 로 0.14 이상에 해당하여 **큰 효과(Large)**로 해석된다. 상권 종류가 매출액 분산의 약 22.8%를 설명한다.
- **결론**: 상권 종류는 2023년 평균 매출액에 유의하고 큰 차이를 만든다. 어떤 상권 종류들 사이에서 차이가 두드러지는지는 분산분석만으로 알 수 없으므로, 구체적 비교를 위해서는 사후검정(`posthoc_oneway`)을 추가로 수행해야 한다.
